### PR TITLE
BREAKING: upgrade angular 16

### PR DESCRIPTION
**Pull request by owilliams320**
## Description

Updgrading Covalent to support Angular 16

### What's included?
- Upgrade support for angular 16.x.x and angular material 16.x.x
- Upgrading the typescript version to **5.1.6**
- Upgrading all NX packages to 16x 
- Code editor project now uses jest over Karma!
- Upgrading material to use the latest version

#### Test Steps

- [ ] `npm run start`
- [ ] then open the browser to localhost:4200 and browse around
- [ ] finally verify the web components storybook works along with all of th components

#### General Tests for Every PR

- [ ] `npm run start` still works.
- [ ] `npm run lint` passes.
- [ ] `npm run stylelint` passes.
- [ ] `npm test` passes and code coverage is not lower.
- [ ] `npm run build` still works.

### CODE REVIEW

File: package.json [hunk 1]
```diff
@@ -27,23 +27,23 @@
...
+    "@angular/animations": "16.x.x",
+    "@angular/cdk": "16.x.x",
+    "@angular/common": "16.x.x",
+    "@angular/compiler": "16.x.x",
+    "@angular/core": "16.x.x",
+    "@angular/forms": "16.x.x",
+    "@angular/material": "16.x.x",
+    "@angular/platform-browser": "16.x.x",
+    "@angular/platform-browser-dynamic": "16.x.x",
+    "@angular/router": "16.x.x",
```
Comment (mikespudwebb) [inline package.json:39]: is this a new syntax?

**Review by mikespudwebb — APPROVED** [review]
Looks straight forward.

The version numbers in package.json seem off to me though. 🤷‍♂️

File: package.json [hunk 2]
```diff
@@ -27,23 +27,23 @@
...
+    "@material/banner": "15.0.0-canary.83355c322.0",
+    "@material/button": "15.0.0-canary.83355c322.0",
+    "@material/card": "15.0.0-canary.83355c322.0",
+    "@material/chips": "15.0.0-canary.83355c322.0",
+    "@material/data-table": "15.0.0-canary.83355c322.0",
+    "@material/icon-button": "15.0.0-canary.83355c322.0",
+    "@material/image-list": "15.0.0-canary.83355c322.0",
```
[the remark this answers is not in the document]
Comment (owilliams320) [reply package.json:46]: oddly they dont have one. This had to match the version that `@angular/material` is pulling

File: package.json (hunk 1 above)
Comment (owilliams320) [reply package.json:39]: Not new syntax its semver. but should help alleviate some of the npm versioning issues we are seeing with folks that consume covalent.

**Pull request merged**